server/src/main/java/org/elasticsearch/TransportVersions.java

```diff
@@ -138,6 +138,7 @@ static TransportVersion def(int id, String uniqueId) {
     public static final TransportVersion V_8_500_072 = def(8_500_072, "e2df7d80-7b74-4afd-9734-aee0fc256025");
     public static final TransportVersion V_8_500_073 = def(8_500_073, "9128e16a-e4f7-41c4-b04f-842955bfc1b4");
     public static final TransportVersion V_8_500_074 = def(8_500_074, "aab0c31c-62d7-4b95-bb29-0b6f367ece64");
+    public static final TransportVersion NESTED_KNN_VECTOR_QUERY_V = def(8_500_999, "nested-knn-vector-query-support");
```

Comment:
@javanna & @jpountz this is a weird one.

We are going to have conflicts on ever main merge. 

Then when we finalize the snapshot, we need to make sure we update this in the PR to merge all the changes into main. 

This is the best "hack" I could think of (irregular name & super future version).

Reply:
This is the suggestion that Ryan made, see https://github.com/elastic/elasticsearch/pull/99312. So eventually we would have the UPGRADE_TO_LUCENE_9_8 there with gap from the previous version to allow new versions to be introduced in `main`, and I would imagine that `NESTED_KNN_VECTOR_QUERY_V` would take the next version.